test/test_main.py

```diff
@@ -0,0 +1,81 @@
+import sys
+from collections import OrderedDict
+sys.path.append('..')
+sys.path.append('./translator/GCP_cloud_function/cloud_function')
+from unittest.mock import MagicMock, patch, call, Mock
+sys.modules['icone_translator'] =Mock()
+import main
+import icone_translator
+import os
+import json
+
+
+@patch('urllib.request')
+def test_get_ftp_file(request):
+    test_url='fake url'
+    test_output=main.get_ftp_file(test_url)
+    request.urlopen.assert_called_with(test_url)
+    assert True
+
+
+@patch('google.cloud.pubsub_v1.PublisherClient')
+@patch.object(main, 'get_ftp_url')
+@patch.object(main, 'get_ftp_file')
+@patch.object(main, 'parse_xml')
+@patch.object(main, 'get_wzdx_schema')
+def test_translate_newest_icone_to_wzdx(get_wzdx_schema, parse_xml, get_ftp_file, get_ftp_url, pubsub):
```

Comment:
This is a good first test with valid data. We need to add additional tests for cases with invalid data. For instance, when `get_wzdx_schema` returns empty or `validate_wzdx` returns false.

Reply:
Added some more edge cases and created a new test function and pushed the updated code to dev.
Thanks

Reply:
Can we add a test to handle the exception case as well?

Reply:
@abinashkonersman We seem to still be missing this.

Reply:
i added another 2 edge cases for no ftp_url  and invalid ftp_url  which handles the exception case

Reply:
Both new functions don't actually call out to the function to test. They just set it up.

Reply:
I called  out to the function i.e main.translate_newest_icone_to_wzdx(None,None) to test